packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -194,5 +192,66 @@ export async function associateAccountToChainArgs(
       'Ethereum linking is not yet supported by this chain'
     )
 
-  return [accountAddress, validTill, proof]
+  return [accountAddress, validUntil, proof]
+}
+
+/**
+ * Wraps the provided challenge according to the key type.
+ *
+ * Ethereum addresses will cause the challenge to be prefixed with
+ * `\x19Ethereum Signed Message:\n` and the length of the message.
+ *
+ * For all other key types the message will be wrapped in `<Bytes>..</Bytes>`.
+ *
+ * @param type The key type that will sign the challenge.
+ * @param challenge The challenge to proof ownership of both account and DID.
+ * @returns The wrapped challenge.
+ */
+export function getWrappedChallenge(
+  type: KeypairType,
```

Comment:
Why can't the type be a union of `ethereum` and `dotsama`?

Reply:
So we can add new types in the future, e.g., Bitcoin, without necessarily breaking the interface.

Reply:
yeah didn't want to create a new type for that. Also i hate `dotsama`. Shouldn't that be `substrate` since there are chains outside of kusama/polkadot that are also compatible?

Reply:
Technically Substrate is the framework, so it would be wrong to indicate with Substrate also chains that implement the Polkadot specification using a different framework. If you don't like dotsama, which I personally do not dislike, what about polkadot?

Reply:
It depends on the meaning you give to the word polkadot. I have seen Joe giving it [this](https://wiki.polkadot.network/docs/learn-polkadot-host) meaning, which would capture all networks you and me are thinking of.

Reply:
[065949a](https://github.com/KILTprotocol/sdk-js/pull/730/commits/065949a9cdb11b8825a6467e57fd31d4f3367a6c)

Reply:
Actually I'm also not really happy with that solution

Reply:
BTW since typescript also uses duck typing, we don't break the interface if we change the type to support more options.